userspace/libsinsp/thread_pool.h

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2024 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <list>
+#include <cstdint>
+#include <cstddef>
+#include <functional>
+#include <memory>
+
+class thread_pool
+{
+public:
+	using routine_id_t = std::function<bool()>*;
```

Comment:
```suggestion
	using routine_id_t = uintptr_t;
```
Since you're doing internal casts already, let's not over-expose internal details in the interface at this point.